lib/rack/multipart/parser.rb

```diff
@@ -486,6 +487,19 @@ def find_encoding(enc)
         Encoding::BINARY
       end
 
+      def handle_dummy_encoding(name, body)
+        # 'dummy' encodings are not properly implemented in Ruby MRI
+        if name.encoding.dummy?
+          # ISO-2022-JP is a legacy but still widely used encoding in Japan
+          # Here we properly convert ISO-2022-JP to UTF-8
+          if name.encoding == Encoding::ISO_2022_JP
+            name = name.encode(Encoding::UTF_8)
+            body = body.encode(Encoding::UTF_8)
+          end
```

Comment:
Why not handle all dummy encodings thsi way, e.g.

```suggestion
          name = name.encode(Encoding::UTF_8)
          body = body.encode(Encoding::UTF_8)
```